CocktailsOrder/CocktailsOrder/Networking/NetworkingV2/NetworkServiceable.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  NetworkServiceable.swift
+//  CocktailsOrder
+//
+//  Created by Cosma Dragos on 14.03.2023.
+//
+
+import Foundation
+
+protocol NetworkServiceable {
+    var session: URLSession { get }
+    var baseURL: String { get }
+}
+
+extension NetworkServiceable {
+
+    /// Main function for calling an endpoint.
+    /// - Parameter endpoint: Provided from a predefined API enumeration.
+    /// - Returns: The decoded model or a custom error.
+    func call<T: Decodable>(endpoint: APICall) async -> Result<T, APIError> {
```

Comment:
this method could be marked with 'throws' and it could return just 'T' directly.
func call<T: Decodable>(endpoint: APICall) async throws -> T

This way, you can directly throw or return the decoded value